### DISCUSSION TITLE
"RIP Redux: Dan Abramov announces future fetcher API"

### DISCUSSION THREAD

**Discussion by tarciozemel** [frontendbr/forum#966] [Discussão]
Agora decidiram que Redux já não é bom o suficiente e algo melhor vai entrar em cena: [https://react-etc.net/entry/rip-redux-dan-abramov-announces-future-fetcher](https://react-etc.net/entry/rip-redux-dan-abramov-announces-future-fetcher)

[Para os mais velhos de guerra, não surpreende.](http://desenvolvimentoparaweb.com/miscelanea/desenvolvimento-orientado-a-modinha-dom/) :grin:

**Comment by NandoSangenetto** [comment]
Parece que tudo foi fruto de uma piada, como você pode ver nesta thread no Twitter:
https://twitter.com/kentcdodds/status/969256012632018944

**Comment by tarciozemel** [comment]
@nandosangenetto Será, será...? haha Pena que você mostrou o tweet rápido demais. =P

**Comment by leandroruel** [comment]
man you suck....

**Comment by bernardodiasc** [comment]
fake news 😂

**Comment by joaoeffting** [comment]
Em pleno 2018 e ainda tem gente chamando React e Redux de modinha?

**Comment by diessica** [comment]
> Para os mais velhos de guerra, não surpreende. 

zzzZZzzzZ....

**Comment by tarciozemel** [comment]
Caramba, pessoal tá sem senso de humor, hein... Aposto que esse povo todo aê indenta com 2 espaços. #paz

**Comment by rflmyk** [comment]
tab com tamanho de 2 também vale? Quando li até acreditei pq no JS morre uma lib nascem dez.

**Comment by rflmyk** [comment]
e é bom lembrar q o BrazilJS Weekly colocou na newsletter, hehehehehehhee.

**Comment by bernardodiasc** [comment]
A notícia faz sentido, não se trata de matar uma tecnologia por outra, se trata de usar uma feature do core pra resolver a gestão de estado global da aplicação. Não há nada, tirando essa notícia falsa, que indique que Redux seria substituído pela API Context, mas se parar pra pensar, faz muito sentindo e é uma ótima ideia.

**Comment by fdaciuk** [comment]
É importante entender também que o Redux nada tem a ver com o React. O que faz o Redux e o React conversarem é o `react-redux`. E adivinha: ele **ja é** uma abstração da Context API ;)

**Comment by devalexandre** [comment]
fiz uns testes com a nova context , achei bem mais simples que o redux,  e até agora não entendi se morreu ou não o react-redux

**Comment by rflmyk** [comment]
Não morreu e acredito q nem vá morrer tão cedo, quem usa redux somente pra passar props para componentes herdeiros, na verdade nem deveria usar, Redux é muito mais do q isso.

**Comment by ninetails** [comment]
[You might not need Redux](https://medium.com/@dan_abramov/you-might-not-need-redux-be46360cf367), by **o próprio** e inclusive recomendado pelo Kent C. Dodds.

Na quinta (ou sexta, meu calendário pessoal está bem bagunçado pelo fato de estar "de férias"), teve uma palestra do KCD no React Dev Summit, ele mesmo falava:
- Use State Components
- Se a aplicação for grande/crescer, use [Unstated](//unstated.io)
- Se a aplicação for muito grande, daí considere utilizar Redux

Mas o KCD também tem um curso sobre [Padrões Avançados em React](https://egghead.io/courses/advanced-react-component-patterns), sua aplicação não necessariamente precisa ter o Redux único pra aplicação, neste curso ele ensinou como utilizar como algo parecido com o novo Context API, achei bem legal ainda mais se for abordado um processo de desenvolvimento orientado a domínio.
